src/Compiler/pars.fsy

```diff
@@ -5709,7 +5708,11 @@ recdExprCore:
   | appExpr
     { let mExpr = rhs parseState 1
       reportParseErrorAt mExpr (FSComp.SR.parsFieldBinding ())
-      Some($1, BlockSeparator.Offside(mExpr.EndRange, None)), [] }
+      // Today SynExpr.Record requires copyInfo: (SynExpr * BlockSeparator) option. 
+      // In this error-recovery branch we want to indicate “this looked like a copy-and-update base expression, 
+      // but there was no WITH and no fields. The dummy BlockSeparator is the smallest way to keep the AST well-formed.
```

Comment:
Have you considered making the separator optional here instead? What are the downsides?